plat/rpi/rpi5/scmi/scmi.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2024 EPAM Systems
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#include <assert.h>
+#include <stdint.h>
+
+#include <drivers/scmi-msg.h>
+#include <drivers/scmi.h>
+
+#include <platform_def.h>
+
+#define RPI_SHM_BASE		RPI_SCMI_SHMEM_BASE
+#define RPI_SHM0_BASE	RPI_SHM_BASE
+
+static struct scmi_msg_channel scmi_channel[] = {
+	[0] = {
+		.shm_addr = RPI_SHM0_BASE,
+		.shm_size = SMT_BUF_SLOT_SIZE,
+	},
+};
+
+struct scmi_msg_channel *plat_scmi_get_channel(unsigned int agent_id)
+{
+	assert(agent_id < ARRAY_SIZE(scmi_channel));
+
+	return &scmi_channel[agent_id];
+}
+
+static const char vendor[] = "EPAM";
+static const char sub_vendor[] = "";
+
+const char *plat_scmi_vendor_name(void)
+{
+	return vendor;
+}
+
+const char *plat_scmi_sub_vendor_name(void)
+{
+	return sub_vendor;
+}
+
+/* Currently supporting Clocks and Reset Domains */
+static const uint8_t plat_protocol_list[] = {
+	SCMI_PROTOCOL_ID_RESET_DOMAIN,
```

Comment:
As of this patch, reset domain is still not supported. Support will be added in the next commit only. Please ensure that commits are self-contained and atomic.

Reply:
Agreed. Also I've removed comment on line 44. It seems to be redundand